lldb/source/Target/StackFrameRecognizer.cpp

```diff
@@ -92,11 +95,13 @@ void StackFrameRecognizerManager::ForEach(
         symbol_name = entry.symbol_regexp->GetText().str();
 
       callback(entry.recognizer_id, entry.recognizer->GetName(), module_name,
-               llvm::ArrayRef(ConstString(symbol_name)), true);
+               llvm::ArrayRef(ConstString(symbol_name)), entry.symbol_mangling,
+               true);
 
     } else {
       callback(entry.recognizer_id, entry.recognizer->GetName(),
-               entry.module.GetCString(), entry.symbols, false);
+               entry.module.GetCString(), entry.symbols, entry.symbol_mangling,
+               false);
     }
   }
 }
```

Comment:
afaict, this code is incomplete. The matching in `GetRecognizerForFrame` also needs to be adjusted.

In https://github.com/llvm/llvm-project/pull/105695, you can find an implementation of pretty much the same idea. However, I implemented only `GetRecognizerForFrame` and skipped, e.g., `ForEach`

Reply:
Feel free to copy-paste / adjust whichever code from #105695 which you might find useful. I will rebase that commit later on, after your changes landed. (I cannot land #105695 before #104523 gets re-applied, anyway)

Reply:
Ah looks like we had the same idea!

Reply:
I'll let you commit your PR instead.

Reply:
I relanded everything in 11d2de436cbab8667fe1f99d7b538e6fb555b4d7.